deploy/Chart/templates/controller/webhook.yaml

```diff
@@ -16,13 +16,9 @@ metadata:
     app.kubernetes.io/name: controller
     app.kubernetes.io/part-of: radius
 data:
-  {{ if $existingSecret }}tls.crt: {{ index $existingSecret.data "tls.crt" }}
-  {{ else }}tls.crt: {{ b64enc $cert.Cert }}
-  {{ end }}
-
-  {{ if $existingSecret }}tls.key: {{ index $existingSecret.data "tls.key" }}
-  {{ else }}tls.key: {{ b64enc $cert.Key }}
-  {{ end }}
+  tls.crt: {{ include "secrets.lookup" (dict "secret" "controller-cert" "namespace" .Release.Namespace "key" "tls.crt" "defaultValue" $cert.Cert) }}
```

Comment:
We do not need to create Secret if secret exists. So we can simplify it like below. 

```
{{- if not $existingSecret }} // ---> UPDATE THIS 
apiVersion: v1
kind: Secret
metadata:
  name: controller-cert
  namespace: {{ .Release.Namespace }}
  labels:
    app.kubernetes.io/name: controller
    app.kubernetes.io/part-of: radius
data:
  tls.crt: {{ b64enc $cert.Cert }}
  tls.key: {{ b64enc $cert.Key }}
{{- end }}
```

Reply:
`secrets.lookup` function checks if there is already a Secret and, if there is, it uses the existing one. Like [here](https://github.com/bitnami/charts/blob/882140cfc5318420fdf563b77201f7ee5f67ca8f/bitnami/rabbitmq-cluster-operator/templates/messaging-topology-operator/validating-webhook-configuration.yaml#L51).

`secrets.lookup` gets the secret name and the namespace that secret should be in, checks if it exists, if it does it returns the existing secret. If not, it returns the default value provided. In this case $cert.Cert, $cert.Key, and $ca.Cert.

Reply:
If we do it this (`{{- if not $existingSecret }}`) way, then Secret gets deleted when we do `--reinstall`. In the `--reinstall` case, the **$existingSecret** is not nil, that part is missing from the template. Let me know if that makes sense. Happy to sync offline.

And I did another test, when we do `--reinstall` Secret doesn't get recreated with this way: https://github.com/radius-project/radius/pull/7022/files#diff-dacf6ba48c5b451f0a3a55b9978b57991a50fb27fc2da1c4785e09e52adc4b09R18-R20.